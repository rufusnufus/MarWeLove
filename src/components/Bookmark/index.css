.bookmark {
    position: absolute;
    top: -25px;
    right: -25px;
    width: 50px;
    height: 50px;
}