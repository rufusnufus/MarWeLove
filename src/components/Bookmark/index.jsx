import React, { useState, useCallback } from 'react'
import { useSelector } from 'react-redux'
import apiService from '../../services/apiService'
import './index.css'

function Bookmark({ bookmark, id }) {
  const [isBookmarked, setIsBookmarked] = useState(bookmark)
  const token = useSelector((state) => state.user.token)

  const onClick = useCallback(
    (event) => {
      event.stopPropagation()
      setIsBookmarked(!isBookmarked)
      apiService.toggleBookmark(id, token)
    },
    [isBookmarked]
  )
  return (
    <img
      src={isBookmarked ? '/images/bookmark.png' : '/images/bookmark_blank.png'}
      alt=""
      onClick={onClick}
      className="bookmark"
    />
  )
}

export default Bookmark
