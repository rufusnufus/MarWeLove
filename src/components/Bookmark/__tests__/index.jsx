/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import Bookmark from '../index'
import mockStore from '../../../__mocks__/store'

let store
beforeEach(() => {
  store = mockStore()
})
afterEach(cleanup)

describe('Bookmark', () => {
  it('Render "off"', () => {
    const { container } = render(
      <Provider store={store}>
        <Bookmark bookmark={false} id="1" />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })

  it('Render "on"', () => {
    const { container } = render(
      <Provider store={store}>
        <Bookmark bookmark id="1" />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })

  it('After click bookmark should be "on"', () => {
    const { container, getByTestId } = render(
      <Provider store={store}>
        <Bookmark bookmark={false} id="1" />
      </Provider>
    )

    expect(container).toMatchSnapshot()
    fireEvent.click(getByTestId('bookmark'))
    expect(container).toMatchSnapshot()
  })

  it('After click bookmark should be "off"', () => {
    const { container, getByTestId } = render(
      <Provider store={store}>
        <Bookmark bookmark id="1" />
      </Provider>
    )

    expect(container).toMatchSnapshot()
    fireEvent.click(getByTestId('bookmark'))
    expect(container).toMatchSnapshot()
  })
})
