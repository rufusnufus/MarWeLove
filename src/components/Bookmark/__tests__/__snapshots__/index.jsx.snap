// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark After click bookmark should be "off" 1`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark.png"
  />
</div>
`;

exports[`Bookmark After click bookmark should be "off" 2`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark_blank.png"
  />
</div>
`;

exports[`Bookmark After click bookmark should be "on" 1`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark_blank.png"
  />
</div>
`;

exports[`Bookmark After click bookmark should be "on" 2`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark.png"
  />
</div>
`;

exports[`Bookmark Render "off" 1`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark_blank.png"
  />
</div>
`;

exports[`Bookmark Render "on" 1`] = `
<div>
  <img
    alt=""
    class="bookmark"
    data-testid="bookmark"
    src="/images/bookmark.png"
  />
</div>
`;
